core/condition.go

```diff
@@ -216,7 +223,28 @@ func isEqual(first, second string) bool {
 			}
 			return false
 		}
+		var isFirstStr, isSecondStr bool
+		if strings.HasPrefix(first, StrFunctionPrefix) && firstHasFunctionSuffix {
+			isFirstStr = true
+			first = strings.TrimSuffix(strings.TrimPrefix(first, StrFunctionPrefix), FunctionSuffix)
+		}
+		if strings.HasPrefix(second, StrFunctionPrefix) && secondHasFunctionSuffix {
+			isSecondStr = true
+			second = strings.TrimSuffix(strings.TrimPrefix(second, StrFunctionPrefix), FunctionSuffix)
+		}
+		if isFirstStr || isSecondStr {
+			return first == second
+		}
 	}
+
+	// If both are numeric, force a numeric comparison.
+	// Ensures valid comparison for numbers expressed in decimal, scientific, etc.
+	firstNumeric, firstParseErr := strconv.ParseFloat(first, 64)
+	secondNumeric, secondParseErr := strconv.ParseFloat(second, 64)
+	if firstParseErr == nil && secondParseErr == nil {
+		return firstNumeric == secondNumeric
+	}
```

Comment:
This comment seems to imply that `1e3` would only be parsed as `1000` if both are numeric. Am I misunderstanding something?

If it `1e3` is always parsed as `1000`, then this isn't a change I can accept right now, because that would technically be a breaking change, even if the `str()` function was added.

Reply:
I think `{ "value": 1e3 }` always gets parsed as 1000. The behavior does change for `{ "value": "1e3" }` (the string `"1e3"`).

If both sides of the comparison came in as actual numbers (which get converted to strings by [jsonpath/Sprintf](https://github.com/TwiN/gatus/blob/0c92534432f4e3bd2561810b6a38d6c1ab6ce67f/jsonpath/jsonpath.go#L61-L63)) or are strings that can be parsed as numbers, they'll trigger [line 245](https://github.com/TwiN/gatus/pull/403/files#diff-c24e9639766cd2b5cdce77b082fa734e67b7d518d7e4b66354ec373bd47e51deR245) right now. If either can't be parsed as a number, the behavior is the same as before and will continue to use string comparison.

The change to handling `"1e3"` and the reduction in range from `int64` to `float64` are probably breaking changes.